listener/src/event_handler/auction_settlement_handler.rs

```diff
@@ -0,0 +1,211 @@
+use super::*;
+
+use failure::Error;
+use slog::Logger;
+use std::sync::Arc;
+
+use dfusion_core::database::DbInterface;
+use dfusion_core::models::Deserializable;
+use dfusion_core::models::util::{PopFromLogData, ToValue};
+use dfusion_core::models::Solution;
+
+use graph::components::ethereum::EthereumBlock;
+use graph::components::store::EntityOperation;
+use graph::data::store::Entity;
+
+use web3::types::{H256, U256};
+use web3::types::{Log, Transaction};
+
+use std::fmt;
+
+use super::EventHandler;
+use super::util;
+
+
+#[derive(Clone)]
+pub struct AuctionSettlementHandler {
+    store: Arc<DbInterface>,
+}
+
+impl AuctionSettlementHandler {
+    pub fn new(store: Arc<DbInterface>) -> Self {
+        AuctionSettlementHandler {
+            store
+        }
+    }
+}
+
+impl Debug for AuctionSettlementHandler {
```

Comment:
Can you not `#[derive(Debug)]` that?

Reply:
It appears I cannot. I would ask you the same for `FluxTransitionHandler`.